test/unit/lib/TopStoriesFeed.test.js

```diff
@@ -1021,6 +1021,18 @@ describe("Top Stories Feed", () => {
       assert.equal(action.type, at.TELEMETRY_PERFORMANCE_EVENT);
       assert.equal(action.data.event, "topstories.domain.affinity.calculation.ms");
     });
+    it("should add idle-daily observer right away, before waiting on init data", async () => {
+      let callOrder = [];
+      instance.addObserver = () => {
+        callOrder.push("add");
+      };
+      instance.removeObserver = () => {
+        callOrder.push("remove");
+      };
+      await Promise.all([instance.onInit(), instance.uninit()]);
+      assert.equal(callOrder[0], "add");
+      assert.equal(callOrder[1], "remove");
```

Comment:
Dunno a better way to test the call order, and in general this was one of the more tricky tests to write, and while the above seems to get the job done, happy to take improvements.

Reply:
Could you do something like 
```
 const initPromise = init();
  assert.calledOnce(Services.obs.addObserver);
  await initPromise;
```